packages/ckeditor5-table/src/tableconfig.ts

```diff
@@ -372,3 +383,35 @@ export interface TableCellPropertiesOptions {
 	 */
 	verticalAlignment?: string;
 }
+
+/**
+ * The configuration of the table layout feature.
+ */
+export interface TableLayoutConfig {
+
+	/**
+	 * The preferred type of the external table to be inserted.
+	 *
+	 * By using this setting you can change the table type detection heuristics.
+	 * The highest priority has recognition by CSS classnames: `content-table` and `layout-table`.
+	 * If the classnames are not present, the default heuristic is used.
+	 *
+	 * When `preferredExternalTableType` is set, the default heuristic is overridden and the table type is set to the value of this setting.
+	 *
+	 * ```ts
+	 * ClassicEditor
+	 * 	.create( {
+	 * 		table: {
+	 * 			tableLayout: {
+	 * 				preferredExternalTableType: 'content' // or 'layout'
+	 * 			}
+	 * 		}
+	 * 	} )
+	 * 	.then( ... )
+	 * 	.catch( ... );
+	 * ```
+	 */
+	preferredExternalTableType: TableType;
+}
+
+export type TableType = 'content' | 'layout';
```

Comment:
The same is declared in the command, maybe we should have a single type for it exported from for example editing?

Reply:
I would keep it in the configuration file, as I have found that these types of files are most commonly used in such cases.

Reply:
Added a [comment](https://github.com/ckeditor/ckeditor5/pull/18109/files#r1998882906) to fix it in the TableTypeCommand PR.